src/main/java/net/snowflake/client/core/FieldSchemaCreator.java

```diff
@@ -0,0 +1,96 @@
+package net.snowflake.client.core;
+
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.Optional;
+import net.snowflake.client.jdbc.BindingParameterMetadata;
+import net.snowflake.client.jdbc.SnowflakeColumn;
+import net.snowflake.client.jdbc.SnowflakeType;
+import net.snowflake.client.log.SFLogger;
+import net.snowflake.client.log.SFLoggerFactory;
+
+@SnowflakeJdbcInternalApi
+public class FieldSchemaCreator {
+  static final SFLogger logger = SFLoggerFactory.getLogger(FieldSchemaCreator.class);
+  public static final int MAX_TEXT_COLUMN_SIZE = 134217728;
+  public static final int MAX_BINARY_COLUMN_SIZE = 8388608;
+
+  public static BindingParameterMetadata buildSchemaForVarchar(
```

Comment:
Maybe we can call it `text`? This is not always a varchar and we operate on logical names here.

Reply:
Ok